tools/tests/test_lint.py

```
@@ -3510,3 +3524,196 @@ def test_a_repeat_does_not_swallow_the_empty_string_finding(tmp_path, monkeypatc
+    )
+    assert exotic not in ignored
+
+
+def test_git_ignored_filters_an_ordinary_tree(tmp_path):
+    """The other polarity: the lawful case still filters, so the fix above is
+    not the filter quietly turning itself off.
+    """
+    subprocess.run(["git", "init", "-q", str(tmp_path)], check=True)
+    (tmp_path / ".gitignore").write_bytes(("skip" + chr(10)).encode("utf-8"))
+    (tmp_path / "skip").mkdir()
+    hidden = tmp_path / "skip" / "a.py"; hidden.write_bytes(b"x = 1" + chr(10).encode())
```

Ruff (0.16.2)

[error] Multiple statements on one line (semicolon)

(E702)